scripts/build/cht-core.yml.template

```diff
@@ -11,11 +11,31 @@ services:
       - "COUCHDB_PASSWORD=${COUCHDB_PASSWORD:-pass}"
       - "COUCHDB_SERVERS=${COUCHDB_SERVERS:-{{ couchdb_servers }}}"
       - "HAPROXY_PORT=${HAPROXY_PORT:-5984}"
+    logging:
```

Comment:
I had a quick look at the failing upgrade e2e test and found this in the logs: 
```
2022-08-29 00:45:38 [31mERROR[39m: Error thrown while installing: StatusCodeError: 500 - {"error":true,"reason":"Invalid docker-compose yml for file cht-core.yml"}
```

It's reporting that this file is not valid. 
This is the file:  https://staging.dev.medicmobile.org/_couch/builds/medic%3Amedic%3Ahaproxy-healthcheck/docker-compose%2Fcht-core.yml
And the validity check is done via: `docker-compose -f .... config`. 

Running the command locally, the error is: 
```services.healthcheck.container_name must be a string```
You have to add this variable in when populating the template here: https://github.com/medic/cht-core/blob/archv3/scripts/build/index.js#L119

Reply:
The variable was defined but the name didn't match. Fixed.

Reply:
I checked the logs for the upgrade e2e test, and it's full of errors across the board that there are no couchdb servers: 
in API (and Sentinel):
```
StatusCodeError: 503 - "<html><body><h1>503 Service Unavailable</h1>\nNo server is available to handle this request.\n</body></html>\n"
    at new StatusCodeError (/api/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/api/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/api/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/api/node_modules/request/request.js:185:22)
    at Request.emit (node:events:527:28)
    at Request.<anonymous> (/api/node_modules/request/request.js:1154:10)
    at Request.emit (node:events:527:28)
    at IncomingMessage.<anonymous> (/api/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (node:events:641:28)
    at IncomingMessage.emit (node:events:539:35) {
  statusCode: 503,
  error: '<html><body><h1>503 Service Unavailable</h1>\n' +
    'No server is available to handle this request.\n' +
    '</body></html>\n'
}
```

In haproxy:
```
[NOTICE] 241/043831 (23) : haproxy version is 2.3.19-0647791
[NOTICE] 241/043831 (23) : path to executable is /usr/local/sbin/haproxy
[ALERT] 241/043831 (23) : backend 'couchdb-servers' has no server available!
```

Reply:
I think it's because on [this line](https://github.com/medic/cht-core/pull/7737/commits/03a3ad52078b44cc18cea1bec5ddf0e289ac295a#diff-54b16304e568a65183de033e3a0864aa4a9ff97ea186674e68a42652c0f0574cR129) we set `couchdb_servers: 'couchdb',` but we're upgrading a clustered setup. So then the new healthcheck code finds that all_nodes.length is not equal to couchdb_servers.length and returns "down". There's some weird logs with the couch containers too, eg:

cht-couchdb.log:
```
Waiting for cht couchdb
couchdb  is ready
Failed to create system database '_users'
Failed to create system database '_replicator'
Failed to create system database '_global_changes'
```

cht-couchdb.2-e2e.log
```
Error: No such container: cht-couchdb.2-e2e
```

Is this left over from when we changed the default to single node?

Reply:
I don't think we upgrade to a cluster, it's still a single node couchdb (we're using the same docker compose files for both). 

> Is this left over from when we changed the default to single node?

The e2e test suite runs over a cluster, but the upgrade test suite runs against single node production compose files. The "log gathering script" is common between the 2 and tries to collect logs from non-existent containers. I fixed that in one of my PRs, but it's just log collecting after a failed test, nothing else.

Reply:
It looks like this is still not working as expected - checking haproxy logs: 
```
[WARNING] 241/084610 (25) : Server couchdb-servers/cht-couchdb is DOWN, reason: Layer7 wrong status, code: 0, info: "via agent : down", check duration: 222ms. 0 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[NOTICE] 241/084610 (25) : haproxy version is 2.3.19-0647791
[NOTICE] 241/084610 (25) : path to executable is /usr/local/sbin/haproxy
[ALERT] 241/084610 (25) : backend 'couchdb-servers' has no server available!
<150>Aug 30 08:46:53 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:53 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:54 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:54 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:55 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:55 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:56 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:56 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:57 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:57 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:58 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:58 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:59 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:46:59 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:47:00 haproxy[25]: 172.18.0.6,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
<150>Aug 30 08:47:00 haproxy[25]: 172.18.0.5,<NOSRV>,503,0,0,0,GET,/,-,admin,'-',222,-1,-,'-'
```

Reply:
Yeah I'm not sure what's going on now. The healthcheck log doesn't say anything useful so I don't know how haproxy is getting the down response.